src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -326,6 +326,11 @@ The .NET Foundation licenses this file to you under the MIT license.
 
   </Target>
 
+  <PropertyGroup Condition="'$(NativeCodeGen)' == 'llvm'">
+    <EmccOptimizationSetting Condition="$(Optimize) == 'true' and $(IlcOptimizationPreference) != 'Size'">-O3</EmccOptimizationSetting>
```

Comment:
Is `-O3` what we want for `-c Release` as the default?

In native compilers it usually implies aggressive speed-to-size tradeoff i. e. larger unrolling thresholds, etc.

What is the size impact on (say) `HelloWasm.wasm` for `-O1/-O2/-O3`?

Reply:
I took off `-g3` at compile and `-g` at link time, `lto` is on and `binaryen` runs.  Added `Oz` for comparison
```
EXEC cannot compile inline asm [E:\GitHub\runtimelab3\src\tests\nativeaot\SmokeTests\HelloWasm\HelloWasm.csproj]
```
|Option | Size |
|--------|-----|
|O1|3,375,405|
|O2|2,952,589|
|O3|2,936,045|
|Oz|2,501,692|

Reply:
So it looks like the right default after all. Interesting that `-Oz` is so much smaller, I wonder why (not necessary to investigate at this point, though).